src/commands/codepush/lib/hash-utils.ts

```diff
@@ -0,0 +1,162 @@
+        });
+    }, Promise.resolve(<void>null));
+
+    generateManifestPromise
+      .then(() => {
+        resolve(new PackageManifest(fileHashesMap));
+      }, reject)
+  })
+}
+
+export async function hashFile(filePath: string): Promise<string> {
+  var readStream: fs.ReadStream = fs.createReadStream(filePath);
+  return hashStream(readStream);
+}
+
+export async function hashStream(readStream: stream.Readable): Promise<string> {
+  return new Promise<string>((resolve, reject) => {
+    var hashStream = <stream.Transform><any>crypto.createHash(HASH_ALGORITHM);
+
+    readStream
+      .on("error", (error: any): void => {
+        hashStream.end();
+        reject(error);
+      })
+      .on("end", (): void => {
+        hashStream.end();
+
+        var buffer = <Buffer>hashStream.read();
+        var hash: string = buffer.toString("hex");
+
+        resolve(hash);
+      });
+
+    readStream.pipe(hashStream);
+  })
+}
+
+export class PackageManifest {
+  private _map: Map<string, string>;
+
+  public constructor(map?: Map<string, string>) {
+    if (!map) {
+      map = new Map<string, string>();
+    }
+    this._map = map;
+  }
+
+  public toMap(): Map<string, string> {
+    return this._map;
+  }
+
+  public async computePackageHash(): Promise<string> {
+    var entries: string[] = [];
+    this._map.forEach((hash: string, name: string): void => {
+      entries.push(name + ":" + hash);
+    });
+
+    // Make sure this list is alphabetically ordered so that other clients
+    // can also compute this hash easily given the update contents.
+    entries = entries.sort();
+
+    return Promise.resolve(
+      crypto.createHash(HASH_ALGORITHM)
+        .update(JSON.stringify(entries))
+        .digest("hex")
+    );
+  }
+
+  public serialize(): string {
+    var obj: any = {};
+
+    this._map.forEach(function (value, key) {
+      obj[key] = value;
+    });
+
+    return JSON.stringify(obj);
+  }
+
+  public static deserialize(serializedContents: string): PackageManifest {
+    try {
+      var obj: any = JSON.parse(serializedContents);
+      var map = new Map<string, string>();
+
+      for (var key of Object.keys(obj)) {
+        map.set(key, obj[key]);
+      }
+
+      return new PackageManifest(map);
+    } catch (e) {
+    }
+  }
+
+  public static normalizePath(filePath: string): string {
+    //replace all backslashes coming from cli running on windows machines by slashes
+    return filePath.replace(/\\/g, "/");
+  }
+
+  public static isIgnored(relativeFilePath: string): boolean {
+    const __MACOSX = "__MACOSX/";
+    const DS_STORE = ".DS_Store";
+    const CODEPUSH_METADATA = ".codepushrelease";
+    return startsWith(relativeFilePath, __MACOSX)
+      || relativeFilePath === DS_STORE
+      || endsWith(relativeFilePath, "/" + DS_STORE)
+      || relativeFilePath === CODEPUSH_METADATA
+      || endsWith(relativeFilePath, "/" + CODEPUSH_METADATA);
+  }
+}
+
+function startsWith(str: string, prefix: string): boolean {
```

Comment:
Use `_.startsWith` from lodash instead of custom implementation.

Reply:
fixed